### DISCUSSION TITLE
ZFS pool monitor

### DISCUSSION THREAD

**Discussion by Serhioromano** [dougiteixeira/proxmoxve#201] [Feature request]
Can we add ZFS pool monitor. If one of the disks goes off, or fail or there is check error, I would like to set some automation. Would that be possible?

**Comment by jeremysherriff** [comment]
ZFS includes a built-in monitoring system ("zed") which may be of interest to you.
The following links will assist in setup and testing:

- https://www.reddit.com/r/homelab/comments/8c09pr/guide_to_setting_up_zed_to_email_alerts_for_zfs/
- https://www.reddit.com/r/zfs/comments/fb8utq/how_to_test_zed_notification_emails/
- https://github.com/openzfs/zfs/issues/4511

I can imagine that setting up monitoring for ZFS via this integration would be difficult, as many proxmox installations do not use ZFS.
